File: week12/ClockAppTemplate/ClockAppTemplate/counter.py
class Counter:
    def __init__(self, name: str):
        self._name = name
        self._count = 0

    @property
    def name(self) -> str:
        return self._name

    @name.setter
    def name(self, value: str):
        self._name = value

    @property
    def ticks(self) -> int:
        return self._count

    def increment(self):
        self._count += 1

    def reset(self):
        self._count = 0

File: week12/ClockAppTemplate/ClockAppTemplate/Counter.cs
﻿using System;
namespace ClockApp
{
    public class Counter
    {
        private int _count;
        public int Ticks
        {
            get { return _count; }
        }

        private string _name;
        public string Name
        {
            get { return _name; }
            set { _name = value; }
        }

        public Counter(string name)
        {
            _name = name;
            _count = 0;
        }

        public void Increment()
        {
            _count++;
        }

        public void Reset()
        {
            _count = 0;
        }
    }
}
